studio/src/editors/merch-card-editor.js

```diff
@@ -181,33 +302,271 @@ class MerchCardEditor extends LitElement {
-        this.dispatchEvent(new CustomEvent('update-fragment', { detail: e }));
+    #updateFragment(event) {
+        const fieldName = event.target.dataset.field;
+        let value = event.target.value || event.detail?.value;
+        value = event.target.multiline ? value?.split(',') : [value ?? ''];
+        this.fragmentStore.updateField(fieldName, value);
+        this.hasChanges = true;
     }
 
-    updateMnemonics(e) {
+    #updateFragmentInternal(event) {
+        const fieldName = event.target.dataset.field;
+        let value = event.target.value;
+        this.fragmentStore.updateFieldInternal(fieldName, value);
+        this.hasChanges = true;
+    }
+
+    #updateMnemonics(event) {
         const mnemonicIcon = [];
         const mnemonicAlt = [];
         const mnemonicLink = [];
-        e.target.value.forEach(({ icon, alt, link }) => {
+        event.target.value.forEach(({ icon, alt, link }) => {
             mnemonicIcon.push(icon ?? '');
             mnemonicAlt.push(alt ?? '');
             mnemonicLink.push(link ?? '');
         });
-        this.fragment.updateField('mnemonicIcon', mnemonicIcon);
-        this.fragment.updateField('mnemonicAlt', mnemonicAlt);
-        this.fragment.updateField('mnemonicLink', mnemonicLink);
-        this.dispatchEvent(new CustomEvent('refresh-fragment'));
+        const fragment = this.fragmentStore.get();
+        fragment.updateField('mnemonicIcon', mnemonicIcon);
+        fragment.updateField('mnemonicAlt', mnemonicAlt);
+        fragment.updateField('mnemonicLink', mnemonicLink);
+        this.fragmentStore.set(fragment);
+        this.hasChanges = true;
+    }
+
+    // #region Toolbar
+
+    aemAction(action, reset = false) {
+        return async function () {
+            this.disabled = true;
+            const ok = await action();
+            if (ok && reset) this.hasChanges = false;
+            this.disabled = false;
+        };
     }
+
+    openFragmentInOdin() {
+        const fragment = this.fragmentStore.get();
+        window.open(
+            `https://experience.adobe.com/?repo=${this.bucket}.adobeaemcloud.com#/@odin02/aem/cf/admin/?appId=aem-cf-admin&q=${fragment?.fragmentName}`,
+            '_blank',
+        );
+    }
+
+    async copyToUse() {
+        const code = `<merch-card><aem-fragment fragment="${Store.fragments.inEdit.get()}"></aem-fragment></merch-card>`;
+        try {
+            await navigator.clipboard.writeText(code);
+            Events.showToast.emit({
+                variant: 'positive',
+                content: 'Code copied to clipboard',
+            });
+        } catch (error) {
+            console.error(`Failed to copy code to clipboard: ${error.message}`);
+            Events.showToast.emit({
+                variant: 'negative',
+                content: 'Failed to copy code to clipboard',
+            });
+        }
+    }
+
+    get toolbar() {
+        return html`<div id="editor-toolbar">
+            <sp-action-group
```

Comment:
The action buttons should not be in merch-card-editor, please bring back editor-panel. :-)